src/main/java/com/epam/aidial/core/controller/DeploymentPostController.java

```diff
@@ -63,17 +64,20 @@ public class DeploymentPostController {
     private final Proxy proxy;
     private final ProxyContext context;
     private final CustomApplicationService applicationService;
-    private final List<BaseFunction<ObjectNode>> enhancementFunctions;
+    private final List<BaseRequestFunction<ObjectNode>> enhancementFunctions;
+
+    private final CollectResponseAttachmentsFn collectResponseAttachmentsFn;
```

Comment:
why do we need this field? you can create instance later

Reply:
Right. I thought we want to avoid of creating an instance multiple times just once. I see it's always created just once in the code.

Reply:
fixed